core/document/src/layers/folder.rs

```diff
@@ -59,36 +59,36 @@ impl Folder {
 	}
 
 	pub fn remove_layer(&mut self, id: LayerId) -> Result<(), DocumentError> {
-		let pos = self.layer_ids.iter().position(|x| *x == id).ok_or(DocumentError::LayerNotFound)?;
+		let pos = self.position_of_layer(id).ok_or(DocumentError::LayerNotFound)?;
```

Comment:
I would probably let the function return an error directly but that is probably personal preference

Reply:
That's what I had at first, but I made the function more generic so it could be used in more places. I'll change it back though if you think it's better.